File: BinaryTree/Leetcode100.java
public class Leetcode100 {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q == null)return true;
        if(p != null && q == null)return false;
        if(p == null && q != null)return false;
        if(p.val != q.val)return false;
        boolean left = isSameTree(p.left,q.left);
        boolean right = isSameTree(p.right,q.right);
        return (left && right);
    }
}

File: BinarySearchTree/Python/Leetcode100.py
class Leetcode100:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if p is None and q is None:
            return True
        if p is None or q is None:
            return False
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right) and p.val == q.val
